### DISCUSSION TITLE
Firmwareupdate 4.0.3

### DISCUSSION THREAD

**Discussion by Michel-MC** [dewenni/ESP_Buderus_KM271#97] [Q&A] [answered]
Hallo,

ich habe das Modul von "the78mole" mit dem Firmwarestand V3.2.4 bei mir im Einsatz. Da mittlerweile schon einige neue Features dazugekommen sind, möchte ich nach Möglichkeit updaten. Ich habe aber leider kein Flash-Kabel dafür, sonst hätte ich das schon gemacht. Es gab da ja eine Änderung der Partitionstabelle...

Nun meine Frage:
Kann ich von V3.2.4 via OTA auf V4.0.3 updaten oder muss das auch zwingend via Kabel geflasht werden wie es in den realeases V3.2.5 oder V3.3.1 angemerkt ist?

Freue mich auf Antworten.

VG Michel

**Comment by the78mole** [comment]
Hallo Michael,

ob es tatsächlich per Kabel geflasht werden muss, weiß ich jetzt nicht 100%, aber ich gehe mal davon aus.

Man benötigt dafür aber kein sonderlich spezielles Flash-Kabel. Es genügt ein USB-seriell-Adapter mit 3,3V-Logik-Pegel und die Flash-Software.
[Dieses Kabel](https://www.reichelt.de/raspberry-pi-usb-zu-ttl-0-9-m-pl2303hx-rpi-usb-ttl-p150567.html) würde funktionieren, die Logikpegel sind 3,3V, allerdings muss muss man mit den 5V Power (rote Leitung) aufpassen, die dürfen nicht an die 3,3V des DEBUG-Steckers angeschlossen werden, sondern entweder an die 5V am Power-Select-Stecker oder man versorgt das Board separat über die Mini-USB-Buchse.
Dann muss man nur etwas auf dem Mäuseklavier (Buttons) spielen und den Flashvorgang starten. Wie das genau geht, habe ich [hier](https://www.youtube.com/watch?v=h_pQlpXaQ1I&t=810s) gezeigt.

Grüße,
Daniel

**Reply by Michel-MC** [reply]
Super, besorge mir jetzt mal das Kabel. 😊Danke Dir. 👍🏻

**Comment by dewenni** [comment]
ja, du wirst es leider nicht ohne so ein Kabel bzw. Adapter machen können.
Es gibt da aber ganz viele Modelle die alle funktionieren sollten.

diese hier sollten auch funktionieren:
[Modell 1](https://www.amazon.de/UART-TTL-Adapter-Konverter-Jumperkabel-kompatibel/dp/B0B7RHPMT7/ref=sr_1_3?__mk_de_DE=ÅMÅŽÕÑ&crid=1FLJ527KVK3BW&dib=eyJ2IjoiMSJ9.hAH0rE1YoY6Rx6GQQn5eWr9xIcJ2MogHPSLtkzbsBP1NfUrYQV0xoUOiVubZytndyz2HPg_0boc1Bba6MLOTQjA1htiHDeUyHU0nHv8T3tqx5DJyg7ICbhlpdzj1zJO6yqnw7E7noZGrqEU6vPbCO3ID3nvBd5ti1dMZDZn0Nyop-czvcTtBaoN0haoU_KQKA5WXtHFSF3WU2_AKl-952GUEMY3lM75jdxxq5Eyjtjc.P-wi2_FjpOzMwswjmsIFK2Yqst-MJ5y2k4scP8LKp28&dib_tag=se&keywords=usb+ttl&qid=1714068672&sprefix=usb+ttl%2Caps%2C97&sr=8-3)

[Modell 2](https://www.amazon.de/AZDelivery-Konverter-kompatibel-Arduino-inklusive/dp/B089QJZ51Z/ref=sr_1_30?__mk_de_DE=ÅMÅŽÕÑ&crid=18SO7D2MIHB39&dib=eyJ2IjoiMSJ9.ucrW6gxS9pJPbY8JOcXEJpB63K8p3tzKhqAGRxIwiIIfTg8k8zniF0XNWe4lJTp_IiMv0_32GrOSthoav2bNxGEhut7mcjV05O5CdIhKnEhN5DNhxASwvI2hXWryiW4Tkq7L1eXe3Au8kaCJYul36C7ZV2CQZ05B5eWW6dvbbOJpde8-eh39kBG0NJwoMb-b.KNZSrqlOTTOtiauB3KzIET38Zhw2bnhexmr7YmRu0cM&dib_tag=se&keywords=esp+adapter&qid=1714068630&sprefix=esp+adapte%2Caps%2C102&sr=8-30)

[Modell 3](https://www.amazon.de/AZDelivery-UART-TTL-Konverter-Jumperkabel-inklusive/dp/B08T24NML9/ref=sr_1_4?__mk_de_DE=ÅMÅŽÕÑ&crid=1FLJ527KVK3BW&dib=eyJ2IjoiMSJ9.hAH0rE1YoY6Rx6GQQn5eWr9xIcJ2MogHPSLtkzbsBP1NfUrYQV0xoUOiVubZytndyz2HPg_0boc1Bba6MLOTQjA1htiHDeUyHU0nHv8T3tqx5DJyg7ICbhlpdzj1zJO6yqnw7E7noZGrqEU6vPbCO3ID3nvBd5ti1dMZDZn0Nyop-czvcTtBaoN0haoU_KQKA5WXtHFSF3WU2_AKl-952GUEMY3lM75jdxxq5Eyjtjc.P-wi2_FjpOzMwswjmsIFK2Yqst-MJ5y2k4scP8LKp28&dib_tag=se&keywords=usb+ttl&qid=1714068672&sprefix=usb+ttl%2Caps%2C97&sr=8-4)

**Comment by Michel-MC** [comment, marked as answer]
Hallo,

noch als Rückinfo:
Ich habe es mit Modell 3 (Jumper auf 3,3V einstellen!!!) und dem Flashtool von Espressif gemacht. Die Kabel habe ich wie in der Doku vom Daniel angeschlossen. Hat auf Anhieb funktioniert und man braucht keine separate Stromversorgung. Danach dann im Setup Mode noch die config.json eingespielt und neu gestartet. War sofort wieder wie gewohnt erreichbar.

Danke Euch nochmals und viele Grüße

Michel
